core/src/smartcontracts/wasm.rs

```diff
@@ -622,6 +701,27 @@ trait ExecuteOperations<'wrld, S: state::GetCommon<'wrld>> {
     }
 }
 
+trait QueryAuthority<'wrld, S> {
+    #[codec::wrap_trait_fn]
+    fn query_authority(state: &S) -> AccountId;
+}
+
+/// Marker trait to have [`QueryAuthority`] default-implemented for concrete [`Runtime`]
+trait DefaultQueryAuthority {}
+
+impl<'wrld, S: state::GetCommon<'wrld>, R: DefaultQueryAuthority> QueryAuthority<'wrld, S> for R {
```

Comment:
```suggestion
impl<'wrld, S: state::GetCommon<'wrld> + HasAuthority> QueryAuthority<'wrld, S> for Runtime<S> {
```

my first reaction is to advise this. Maybe you can't do it for some reason?